sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -6695,6 +6695,18 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val MERGE_INTO_SOURCE_NESTED_TYPE_UPDATE_BY_FIELD =
+    buildConf("spark.sql.merge.nested.type.assign.by.fieldv2")
```

Comment:
Hi, @szehon-ho . The naming space design looks weird. Why `nested` is at the different level like the following?

```
spark.sql.merge.nested.type.assign.by.fieldv2
spark.sql.merge.source.nested.type.coercion.enabled
```